module/spl/spl-condvar.c

```diff
@@ -231,7 +231,7 @@ EXPORT_SYMBOL(__cv_timedwait_sig);
  * 'expire_time' argument is an absolute clock time in nanoseconds.
  * Return value is time left (expire_time - now) or -1 if timeout occurred.
  */
-static clock_t
+static hrtime_t
 __cv_timedwait_hires(kcondvar_t *cvp, kmutex_t *mp, hrtime_t expire_time,
```

Comment:
This bugs me too but the illumos version of this does return a `clock_t`.  However, it looks like have a longer standing bug where we're not converting the hrtime_t to a clock_t with `NSEC_TO_TICK().`